### PR TITLE
Java 17 code style

### PR DESCRIPTION
**Pull request by christopher-cudennec**

### CODE REVIEW

File: sda-commons-client-jersey/src/test/java/org/sdase/commons/client/jersey/oidc/OidcRequestFilterTest.java [hunk 1]
```diff
@@ -147,9 +147,10 @@ private void setupTokenEndpoint(boolean withBody) {
 
     if (withBody) {
       response.withBody(
-          "{\"access_token\": \"xxxx.yyyy.zzzz\",\n"
-              + "  \"expires_in\": 60,\n"
-              + "  \"token_type\": \"Bearer\"}");
+          """
+            {"access_token": "xxxx.yyyy.zzzz",
+              "expires_in": 60,
+              "token_type": "Bearer"}""");
```
Comment (JoergSiebahn) [inline sda-commons-client-jersey/src/test/java/org/sdase/commons/client/jersey/oidc/OidcRequestFilterTest.java:153]: NIT:

wouldn't it be nicer to format this more readable?

```suggestion
          """
            {
              "access_token": "xxxx.yyyy.zzzz",
              "expires_in": 60,
              "token_type": "Bearer"
            }
          """);
```

Comment (JoergSiebahn) [reply sda-commons-client-jersey/src/test/java/org/sdase/commons/client/jersey/oidc/OidcRequestFilterTest.java:153]: Same for a few other multiline Jsons.

File: sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/AuthBundle.java [hunk 2]
```diff
@@ -143,25 +143,25 @@ protected org.apache.hc.client5.http.impl.classic.HttpClientBuilder createBuilde
...
+    return switch (keyLocation.getType()) {
+      case PEM -> new PemKeySource(
+          keyLocation.getPemKeyId(),
+          keyLocation.getPemSignAlg(),
+          keyLocation.getLocation(),
+          keyLocation.getRequiredIssuer());
+      case OPEN_ID_DISCOVERY -> {
         validateKeyLocation(keyLocation.getLocation(), keyLocation.getRequiredIssuer());
-        return new OpenIdProviderDiscoveryKeySource(
+        yield new OpenIdProviderDiscoveryKeySource(
```
Comment (JoergSiebahn) [inline sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/AuthBundle.java:154]: 🤯

File: sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/JwksKeySource.java [hunk 3]
```diff
@@ -120,32 +120,25 @@ private boolean isSupportedAlg(Key key) {
...
-        return "secp256r1";
-      case "P-384":
-        return "secp384r1";
-      case "P-521":
-        return "secp521r1";
-      default:
-        throw new KeyLoadFailedException(
-            "EC keys are supported but loaded an unsupported EC curve: '" + crv + "'");
-    }
+    return switch (crv) {
```
Comment (JoergSiebahn) [inline sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/JwksKeySource.java:123]: 😲

**Review by JoergSiebahn — APPROVED** [review]
Good amount of code style changes for one PR.

File: sda-commons-client-jersey/src/test/java/org/sdase/commons/client/jersey/oidc/OidcRequestFilterTest.java (hunk 1 above)
Comment (christopher-cudennec) [reply sda-commons-client-jersey/src/test/java/org/sdase/commons/client/jersey/oidc/OidcRequestFilterTest.java:153]: I mostly used IntelliJ's automatic actions to convert code. But you're right, I can do it better manually. I will go through the multiline changes.

**Review by JoergSiebahn — APPROVED** [review]

**Pull request merged**